tests/wallet/nft_wallet/test_nft_wallet.py

```diff
@@ -1308,6 +1308,132 @@ async def test_nft_bulk_set_did(self_hostname: str, two_wallet_nodes: Any, trust
+
+    await time_out_assert(30, wallet_is_synced, True, wallet_node_0, full_node_api)
+    res = await api_0.create_new_wallet(dict(wallet_type="nft_wallet", name="NFT WALLET 1", did_id=hmr_did_id))
+    assert isinstance(res, dict)
+    assert res.get("success")
+    nft_wallet_0_id = res["wallet_id"]
+    res = await api_0.create_new_wallet(dict(wallet_type="nft_wallet", name="NFT WALLET 2"))
+    assert isinstance(res, dict)
+    assert res.get("success")
+    nft_wallet_1_id = res["wallet_id"]
+    await time_out_assert(30, did_wallet.get_confirmed_balance, 1)
+
+    # Create a NFT with DID
+    resp = await api_0.nft_mint_nft(
+        {
+            "wallet_id": nft_wallet_0_id,
+            "hash": "0xD4584AD463139FA8C0D9F68F4B59F185",
+            "uris": ["https://www.chia.net/img/branding/chia-logo.svg"],
+            "mu": ["https://www.chia.net/img/branding/chia-logo.svg"],
+            "did_id": hmr_did_id,
+        }
+    )
+    sb = await make_new_block_with(resp, full_node_api, ph)
+    # ensure hints are generated
+    assert compute_memos(sb)
+    await wait_rpc_state_condition(
+        30, api_0.nft_get_nfts, [{"wallet_id": nft_wallet_0_id}], lambda x: len(x["nft_list"]) > 0
+    )
+    resp = await api_0.nft_mint_nft(
+        {
+            "wallet_id": nft_wallet_1_id,
+            "hash": "0xD4584AD463139FA8C0D9F68F4B59F186",
+            "uris": ["https://www.chia.net/img/branding/chia-logo.svg"],
+            "mu": ["https://www.chia.net/img/branding/chia-logo.svg"],
+            "did_id": "",
+        }
+    )
+    sb = await make_new_block_with(resp, full_node_api, ph)
+    # ensure hints are generated
+    assert compute_memos(sb)
+
+    # Check DID NFT
+    coins_response = await wait_rpc_state_condition(
+        30, api_0.nft_get_nfts, [{"wallet_id": nft_wallet_0_id}], lambda x: len(x["nft_list"]) == 1
+    )
+    coins = coins_response["nft_list"]
+    nft1 = coins[0]
+    assert len(coins) == 1
+    assert coins[0].owner_did is not None
+    coins_response = await wait_rpc_state_condition(
+        30, api_0.nft_get_nfts, [{"wallet_id": nft_wallet_1_id}], lambda x: len(x["nft_list"]) == 1
+    )
+    coins = coins_response["nft_list"]
+    nft2 = coins[0]
+    assert len(coins) == 1
+    assert coins[0].owner_did is None
+    nft_coin_list = [
+        {"wallet_id": nft_wallet_0_id, "nft_coin_id": nft1.nft_coin_id.hex()},
+        {"wallet_id": nft_wallet_1_id, "nft_coin_id": nft2.nft_coin_id.hex()},
+    ]
+    resp = await api_0.nft_transfer_bulk(dict(target_address=address, nft_coin_list=nft_coin_list))
+    sb = await make_new_block_with(resp, full_node_api, ph)
+    # ensure hints are generated
+    assert compute_memos(sb)
+    await time_out_assert(30, get_wallet_number, 2, wallet_node_1.wallet_state_manager)
+    coins_response = await wait_rpc_state_condition(
+        30, api_1.nft_get_nfts, [{"wallet_id": 2}], lambda x: len(x["nft_list"]) == 2
+    )
+    coins = coins_response["nft_list"]
```

Comment:
you should check that correct coins are transferred over and also that DID is reset